autotx/tests/agents/token/research/test_research.py

```diff
@@ -1,119 +1,162 @@
-from crewai import Task, Agent
+import os
+from textwrap import dedent
 import pytest
+from autotx.agents import ResearchTokensAgent
 from autotx.agents.ResearchTokensAgent import (
-    ResearchTokensAgent,
     filter_token_list_by_network,
     get_coingecko,
 )
+from autogen import AssistantAgent, UserProxyAgent, GroupChatManager
+import autogen
+
+from autotx.utils.constants import OPENAI_API_KEY, OPENAI_MODEL_NAME
 from autotx.utils.ethereum.networks import ChainId
 
+@pytest.fixture()
+def user_proxy_agent() -> UserProxyAgent:
+    llm_config = { "cache_seed": None, "config_list": [{"model": OPENAI_MODEL_NAME, "api_key": OPENAI_API_KEY}]}
+    user_proxy_agent = UserProxyAgent(
```

Comment:
The creation of the `UserProxyAgent` and `GroupoChatManager` is done in 2 places, utilizing identical construction parameters which include non-trival & likely to change prompt strings. I think we need to have a unified place where we construct these agents, which gets used in tests and the production app.